e2e/e2e.go

```diff
@@ -299,16 +301,37 @@ func main() {
 		log.Fatal().Err(err).Msg("Failed to run devconfig")
 	}
 
-	log.Info().Msg("Starting devenv tunnel")
-	cmd = exec.CommandContext(ctx, "devenv", "--skip-update", "tunnel")
-	cmd.Stderr = os.Stderr
-	cmd.Stdout = os.Stdout
-	cmd.Stdin = os.Stdin
-	err = cmd.Start()
-	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to start devenv tunnel")
+	var killLocalizer bool
+	if !localizer.IsRunning() {
+		killLocalizer = true
+
+		log.Info().Msg("Starting devenv tunnel")
+		cmd = exec.CommandContext(ctx, "devenv", "--skip-update", "tunnel")
+		cmd.Stderr = os.Stderr
+		cmd.Stdout = os.Stdout
+		cmd.Stdin = os.Stdin
+		err = cmd.Start()
+		if err != nil {
+			log.Fatal().Err(err).Msg("Failed to start devenv tunnel")
+		}
+
+		tick := time.NewTicker(2 * time.Second)
```

Comment:
you can simplify this a lot by using [async.Sleep](https://github.com/getoutreach/async/blob/fbcac8debd5ae5fa41e887f3de7133cc23ac2244/pkg/async/async.go#L132)

```golang
for ctx.Err() != nil && !localize.IsRunning() {
     async.Sleep(ctx, time.Second*2)
}
```